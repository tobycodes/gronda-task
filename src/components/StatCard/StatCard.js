import React from "react";

const StatCard = ({ statTitle, currentPeriod, lastPeriod, bgColors }) => {
	return (
		<div
			className="card"
			style={{
				backgroundImage: `linear-gradient(to right, ${bgColors?.first}, ${bgColors?.second})`,
			}}
		>
			<span className="card__sub">{statTitle}</span>
			<span className="card__value">{currentPeriod}</span>
			<span className="card__sub">{`${lastPeriod} Last period`}</span>
		</div>
	);
};

export default StatCard;
