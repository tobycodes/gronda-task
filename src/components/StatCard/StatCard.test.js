import React from "react";
import { shallow, mount } from "enzyme";

import StatCard from "./StatCard";

const props = {
	currentPeriod: 45,
	lastPeriod: 30,
	bgColor: { first: "#F5C099", second: "#F29E98" },
};

describe("<StatCard />", () => {
	it("expects to render a StatCard component", () => {
		expect(shallow(<StatCard {...props} />)).toMatchSnapshot();
	});

	it("expects to render 3 span elements", () => {
		const wrapper = shallow(<StatCard {...props} />);

		expect(wrapper.find("span").length).toEqual(3);
	});

	it("expects to receive correct stat data and background color props", () => {
		const wrapper = mount(<StatCard {...props} />);

		expect(wrapper.props().currentPeriod).toEqual(45);
		expect(wrapper.props().lastPeriod).toEqual(30);
		expect(wrapper.props().bgColor).toMatchObject(props.bgColor);
	});
});
