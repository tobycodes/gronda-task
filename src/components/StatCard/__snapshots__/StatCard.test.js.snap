// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatCard /> expects to render a StatCard component 1`] = `ShallowWrapper {}`;
